### DISCUSSION TITLE
About custom firmware

### DISCUSSION THREAD

**Discussion by tianzpxyz** [xyzroe/XZG#129] [General]
Excuse me sir, how can I customize the firmware compiled based on my own hardware, and modify those places? Thank you for your answer. At present, I use the esp32 development version to brush your firmware without a model, the hardware version is ESP32-D0WD-V3, I tried to connect the Zigbee module, but it cannot be detected. How can I make it work, thanks.

**Comment by xyzroe** [comment]
You need to add you hardware config to [this](https://github.com/xyzroe/XZG/blob/main/src/const/hw.cpp) file.
